sdk/src/accounts/webSocketDriftClientAccountSubscriber.ts

```diff
@@ -132,9 +153,75 @@ export class WebSocketDriftClientAccountSubscriber
 		this.isSubscribed = true;
 		this.subscriptionPromiseResolver(true);
 
+		// delete initial data
+		this.removeInitialData();
+
 		return true;
 	}
 
+	async setInitialData(): Promise<void> {
+		const connection = this.program.provider.connection;
+
+		if (!this.initialPerpMarketAccountData) {
+			const perpMarketPublicKeys = this.perpMarketIndexes.map((marketIndex) =>
+				getPerpMarketPublicKeySync(this.program.programId, marketIndex)
+			);
+			const perpMarketAccountInfos = await connection.getMultipleAccountsInfo(
+				perpMarketPublicKeys
+			);
+			this.initialPerpMarketAccountData = new Map(
+				perpMarketAccountInfos.map((accountInfo) => {
+					const perpMarket = this.program.coder.accounts.decode(
+						'PerpMarket',
+						accountInfo.data
+					);
+					return [perpMarket.marketIndex, perpMarket];
+				})
+			);
+		}
+
+		if (!this.initialSpotMarketAccountData) {
+			const spotMarketPublicKeys = this.spotMarketIndexes.map((marketIndex) =>
+				getSpotMarketPublicKeySync(this.program.programId, marketIndex)
+			);
+			const spotMarketAccountInfos = await connection.getMultipleAccountsInfo(
+				spotMarketPublicKeys
+			);
+			this.initialSpotMarketAccountData = new Map(
+				spotMarketAccountInfos.map((accountInfo) => {
+					const spotMarket = this.program.coder.accounts.decode(
+						'SpotMarket',
+						accountInfo.data
+					);
+					return [spotMarket.marketIndex, spotMarket];
+				})
+			);
+		}
+
+		const oracleAccountInfos = await connection.getMultipleAccountsInfo(
+			this.oracleInfos.map((oracleInfo) => oracleInfo.publicKey)
+		);
+		this.initialOraclePriceData = new Map(
+			this.oracleInfos.map((oracleInfo, i) => {
+				const oracleClient = this.oracleClientCache.get(
+					oracleInfo.source,
+					connection,
+					this.program
+				);
+				const oraclePriceData = oracleClient.getOraclePriceDataFromBuffer(
+					oracleAccountInfos[i].data
+				);
+				return [oracleInfo.publicKey.toString(), oraclePriceData];
+			})
+		);
+	}
+
+	removeInitialData() {
+		delete this['initialPerpMarketAccountData'];
+		delete this['initialSpotMarketAccountData'];
+		delete this['initialOraclePriceData'];
```

Comment:
small nit, but i'd prefer to set to null or clear the Map rather than removing off the object using delete. should be equivalent space savings and i think is generally safer for the future and more idiomatic